src/interface_rpc_url_provider.py

```diff
@@ -11,7 +11,12 @@
 logger = logging.getLogger(__name__)
 
 class RpcUrlProvider(Object):
-    """RPC url provider interface."""
+    """
+    RPC url provider interface.
+
+    This interface is used by relay chain clients broadcast their rpc urls to parachain clients
+    that use them for relay-over-rpc.
```

Comment:
Nit: no consistency with capitalization of abbreviations (in more than this place)

```suggestion
    RPC URL provider interface.
    This interface is used by relay chain clients broadcast their RPC URLs to parachain clients
    that use them for relay-over-rpc.
```